### ISSUE TITLE
Comments and Suggestions on Pluto and SDR Sharp

### ISSUE THREAD

**Issue by RichardKCollins** [Manawyrm/sdrsharp-plutosdr#23]
Thank you for providing a way to use the Pluto with SDR#.  It was ugly and dirty, but I got it to work.  Here are some things to consider.

Your Filenames are not distinguishable once they are inside the SDR# installation folder.  Could you prefix them so they are clearly all Pluto files?  For instance:

SDRSharp.PlutoSDR.dll - this follows their conventions and you had to name it this.  But it could be named anything, if you change the FrontEnds.xml instructions
  <add key="PlutoSDR" value="SDRSharp.PlutoSDR.PlutoSDRIO,SDRSharp.PlutoSDR" />
I was confused because the SDRSharp.PlutoSDR.PlutoSDRIO is not a file, but the driver that has to be installed separately?

Here is where I would start:

SDRSharp.PlutoSDR.dll  --> PlutoSDR.SDRSharp.dll
libad9361.dll                  --> PlutoSDR.libad9361.dll
libiio.dll                          --> PlutoSDR.libiio.dll
libiio-sharp.dll                --> PlutoSDR.libiio-sharp.dll
libserialport-0.dll            --> PlutoSDR.libserialport-0.dll
libusb-1.0.dll                   --> PlutoSDR.libusb-1.0.dll
libxml2.dll                        --> PlutoSDR.libxml2.dll
msvcr120.dll                    --> PlutoSDR.msvcr120.dll

Everyone is duplicating certain functions.  I cannot tell if they are true duplicates or just all using the same libusb functions.

I have RTL_SDR V3s, AirSpy Discovery, generic RTL_SDR, Nooelec E4000, Adalm-Pluto.  I expect I will have more.  I like SDR# to view the receiver. The gains are ALL different. The filters are ALL different. 

I am using rtl_power on the RTL devices to scan their entire frequency range.  And probably will use rx_power, the Soapy version, to cover Pluto and AirSpy and others.

I am trying to develop a complete map of the global electromagnetic field.  Particularly, from microHz to TeraHz.  That means using a whole range of devices.  Having standards will allow individual devices to be calibrated.  I am buying pairs to check manufacturing variation.

I have to have all the sensors individually calibrated, and to be able to compare pairs or any number of sensors with each other, and to compare the whole of the SDR network with the magnetometer, radio astronomy and other electromagnetic arrays. I will provide an index and a database soon, so people can see what I find.  If you are interested in this sort of thing, I am happy to explain further.  

The instructions on setting up a Pluto are bad, all the way from Analog.com to you.  There are too many out of date and fragmentary pieces by too many authors.  

The Soapy also has a way to handle dlls required by different SDR software. They are doing an OK job, but everyone is going in different directions, on different sites, with different quality and approach to instructions and names. 

Richard Collins, The Internet Foundation

**Comment by RichardKCollins** [comment]
I have no real experience with radios.  I know what I want to do, and what needs to be done. But I do not know what is the truth in terms of what signals are there and what each of the SDRs should be seeing.  I tried to open two SDR# sessions with Pluto in one and an RTL V3 in the other.  I have done this with two RTL V3s and an E4000 no problems.  Here when I try to switch back and forth, in the same session, from RTL to Pluto and back it crashed badly.  So I think you might not be closing things properly, or they are not.

I am attaching an image.  The gains of the Pluto and SDR are considerably different.  I have a pigtail with both devices listening to the same antenna (an RTL_SDR V3 antenna)  I do not know if this is noise, or real signals. But they are similar. IF I could store the FFTs from both, I can statistically compare them to get a better idea. IF I COULD control the gains, the starting frequency, width and sampling time precisely, then I can try to match the two signals in real time.  Over a few MHz it should be possible.  You have the Pluto running at 4 Msps, and the RTL_SDR V3 is running at 3 Msps.

**Comment by RichardKCollins** [comment]
I am not sure where to write here.  Do I "reopen and comment" or "comment"? Wish me luck.

I wanted to show you how hard it is to match a Pluto and an RTL_SDR V3 running on the same antenna.  Their gains are related, just not by the numbers used in the programs.  Here I managed to get two SDR#s running - one with a Pluto RSP1A and one with an RTL_SDR V3.  There is no way to simple set the start and width of each display.  It is ALL unlabeled (no numeric indicator) sliders that are VERY hard to move precisely.  There is no way, that I found, to lock the center frequency.  They can each cover 3 MHz wide.  And if I am zooming, they should both zoom the same.  I think it best if one display could show several SDRs looking at exactly the same frequency region.

I DO NOT need the center frequency at all.  It only tells where to pull out audio signals.  So I could drop that panel and just have the FFT display.

![Compare Pluto and RTL_SDR Near 503 MHz](https://user-images.githubusercontent.com/25554280/65475430-210b2900-de45-11e9-8ccf-f2f3e2ec4dec.PNG)

**Comment by RichardKCollins** [comment]
Here is the same region with the same RTL_SDR V3, but an "NooElec NESDR SMArTee".  I can get them close, but the gains are different. The software could fix that in a few lines.  Tedious.

I tried to do a comparison with the AirSpy Discovery, but they locked it to AutoGain.  I talked to them about that, but they seem to feel "why should we bother, our autogain algorithm is perfect for everyone in every situation. Why let people set their own RF gain?"  So it only shows flat nothing.  The RTL_SDR's are the most reliable. If I see things going on in an RTL_SDR, it is probably closer to the truth.  They are not getting anywhere near their limits, if they would use the FFT data as numbers, rather than eye candy.

![Compare RTL_SDR and E4000 near 503 MHz](https://user-images.githubusercontent.com/25554280/65478249-ac3cec80-de4e-11e9-8f39-5846b6d78744.PNG)
